src/taskManager/Task.java

```diff
@@ -0,0 +1,89 @@
+package taskManager;
+
+public class Task implements Taskable {
+    private String title;
+    private String description;
+    private String priority;
+
+    //use Enum for status
+    private String status;
+
+    public Task(final String title, final String description, final String priority, final String status) {
+        this.title = title;
+        this.description = description;
+        this.priority = priority;
+        this.status = status;
+    }
+
+    public void setTitle(final String title) {
+        if (title.equals("")) {
+            System.out.println("Empty values are not allowed");
+            return;
+        }
+        this.title = title;
+    }
+
+    public void setDescription(final String description) {
+        if (description.equals("")) {
+            System.out.println("Empty values are not allowed");
+            return;
+        }
+        this.description = description;
+    }
+
+    public void setPriority(final String priority) {
+        if (priority.equals("")) {
+            System.out.println("Empty values are not allowed");
+            return;
+        }
+        this.priority = priority;
+    }
+
+    public void setStatus(final String status) {
+        if (status.length() > 1) {
+            this.status = status;
+            return;
+        }
+        System.out.println("Status should contain at least 2 characters");    }
```

Comment:
to be consistent with the previous setter:
```suggestion
        if (status.length() < 2) {
            System.out.println("Status should contain at least 2 characters");
            return;
        }
        this.status = status;
    }
```

Reply:
Yeah, I agree that the validation part should come first. Thanks!